### PR TITLE
Give access to reading/writing the scratchpad.

### PR DESCRIPTION
**Pull request by floitsch**
Also allow to (re)store the scratchpad to/from EEPROM.

### CODE REVIEW

File: examples/scratchpad.toit [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  // Use the low and high temperature registers as external memory.
+  scratchpad := ds18b20.read_scratchpad
+
+  // Since we don't want to change the resolution of the sensor, we only
+  // change the alarm registers.
+  ds18b20.write_scratchpad #[
+    0x12,
+    0x34,
+    scratchpad[4],  // The resolution is the 5th byte of the read bytes.
```
Comment (erikcorry) [inline examples/scratchpad.toit:23]:
```suggestion
    scratchpad[4],  // The resolution is in position 4 of the read bytes.
```

File: src/ds18b20.toit [hunk 2]
```diff
@@ -142,12 +149,174 @@ class Ds18b20:
...
+
+  The scratchpad is the DS18B20's memory. It contains the temperature, the
+    alarm temperatures, the configuration register, and the CRC. Unless
+    committed to EEPROM, the scratchpad is volatile and is reset to the
+    EEPROM values on power up.
+
+  If $check_crc is true, then verifies the CRC of the scratchpad (the last byte).
+
+  The result contains the following bytes:
+  - 0: temperature LSB
```
Comment (erikcorry) [inline src/ds18b20.toit:172]: I think these should be named constants.
HIGH_ALARM_READ_INDEX ::= 2
HIGH_ALARM_WRITE_INDEX ::= 0

File: tests/esp32/scratchpad.toit [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  high := random 255
+  low := random 255
+  device.write_scratchpad #[high, low, resolution_register_value]
+
+  // Read the values back.
+  scratchpad = device.read_scratchpad
+  expect_equals 9 scratchpad.size
+  expect_equals high scratchpad[2]
+  expect_equals low scratchpad[3]
+  // The ds18b20 forces the configuration bit 8 to 0, and bits 0-4 to 1.
```
Comment (erikcorry) [inline tests/esp32/scratchpad.toit:50]:
```suggestion
  // The DS18B20 forces the configuration bit 8 to 0, and bits 0-4 to 1.
```

File: src/ds18b20.toit [hunk 4]
```diff
@@ -142,12 +149,174 @@ class Ds18b20:
...
+      --high_alarm_temperature/float
+      --low_alarm_temperature/float
+      --resolution/int=RESOLUTION_12_BITS
+      --commit/bool=false:
+    high_int := high_alarm_temperature.round.to_int
+    low_int := low_alarm_temperature.round.to_int
+    if not -128 <= high_int <= 127: throw "HIGH ALARM TEMPERATURE OUT OF RANGE"
+    if not -128 <= low_int <= 127: throw "LOW ALARM TEMPERATURE OUT OF RANGE"
+    if not RESOLUTION_9_BITS <= resolution <= RESOLUTION_12_BITS: throw "INVALID_ARGUMENT"
+    config_value := (12 - resolution) << 5
```
Comment (erikcorry) [inline src/ds18b20.toit:284]: This looks wrong.  RESOLUTION_9_BITS should give a config of 0 << 5

**Review by erikcorry — APPROVED** [review]

File: src/ds18b20.toit (hunk 2 above)
Comment (floitsch) [reply src/ds18b20.toit:181]: done.

File: src/ds18b20.toit (hunk 4 above)
Comment (floitsch) [reply src/ds18b20.toit]: oops.
done.

**Pull request merged**